### DISCUSSION TITLE
await tasks on task completion

### DISCUSSION THREAD

**Discussion by GiuseppeDiGuglielmo** [cocotb/cocotb#3590] [Q&A] [answered]
Given a few tasks (`start_soon()`) what is the proper way to `await` for their completion?

Since `fork()` is getting deprecated the following solution doesn't seem an option anymore:

```
@cocotb.test()
def test(dut):
    t1 = cocotb.fork(Thread(dut))
    t2 = cocotb.fork(Thread(dut))
    t3 = cocotb.fork(Thread(dut))
    yield [t1.join(), t2.join(), t3.join()]
```

**Comment by ktbarrett** [comment, marked as answer]
Use `cocotb.triggers.First`.

```python
await First(t1, t2, t3)
```

**Reply by GiuseppeDiGuglielmo** [reply]
This helped!

BTW,
```python
await Combine(t1, t2, t3)
```
was what I actually needed 
https://docs.cocotb.org/en/stable/triggers.html#cocotb.triggers.Combine
